source/common/logger/ur_logger.hpp

```diff
@@ -78,37 +78,47 @@ inline Logger create_logger(std::string logger_name) {
     std::unique_ptr<logger::Sink> sink;
 
     env_var_name << "UR_LOG_" << logger_name;
-    auto map = getenv_to_map(env_var_name.str().c_str());
-    if (!map.has_value()) {
-        return Logger(std::make_unique<logger::StderrSink>(logger_name));
-    }
-
     try {
+        auto map = getenv_to_map(env_var_name.str().c_str());
+        if (!map.has_value()) {
+            return Logger(std::make_unique<logger::StderrSink>(logger_name));
+        }
+
         auto kv = map->find("level");
         if (kv != map->end()) {
             auto value = kv->second.front();
             level = str_to_level(value);
+            map->erase(kv);
         }
 
         kv = map->find("flush");
         if (kv != map->end()) {
             auto value = kv->second.front();
             flush_level = str_to_level(value);
+            map->erase(kv);
         }
 
         std::vector<std::string> values = {default_output};
         kv = map->find("output");
         if (kv != map->end()) {
             values = kv->second;
+            map->erase(kv);
+        }
+
+        if (!map->empty()) {
+            std::cerr << "Wrong logger environment variable parameter: '"
+                      << map->begin()->first
+                      << "'. Default logger options are set.";
+            return Logger(std::make_unique<logger::StderrSink>(logger_name));
         }
 
         sink = values.size() == 2
                    ? sink_from_str(logger_name, values[0], values[1])
                    : sink_from_str(logger_name, values[0]);
     } catch (const std::invalid_argument &e) {
-        std::cerr
-            << "Error when creating a logger instance from environment variable"
-            << e.what();
+        std::cerr << "Error when creating a logger instance from the '"
+                  << env_var_name.str() << "' environment variable:\n"
+                  << e.what() << std::endl;
         return Logger(std::make_unique<logger::StderrSink>(logger_name));
     }
```

Comment:
can't the rest of the function throw (e.g., `Logger(level, std::move(sink));`) ?

Reply:
I assumed it cannot throw as the only possibility of this ctor to do this is when there is no sink. I think that there is guarantee to have a valid sink at this point of the function, since the exception would be already caught when calling `sink_from_str()`.
If I were to check for exceptions in the return statement, I would have to make the whole `create_logger()` funtion a throwable one, and this is sth I wanted to avoid here by making sure that the default logger is properly created.